collectors/lib/kafka_offset_collector.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+
+import itertools
+import time
+import json
+import logging
+
+from kazoo.client import KazooClient
+from kazoo.exceptions import NoNodeError
+from kafka import KafkaClient
+from kafka.common import OffsetRequest, LeaderNotAvailableError
+
+from collectors.lib.optimizely_utils import format_tsd_key
+
+# TSD UTILITIES
+
+
+class Metric(object):
+    LAG_METRIC = 'kafka.consumer.lag'
+    CONSUMER_OFFSET_METRIC = 'kafka.consumer.offset'
+    TOPIC_SIZE_METRIC = 'kafka.broker.offset'
+
+
+class KafkaPaths(object):
+    KAFKA_CHROOT = None
+
+    @classmethod
+    def topic_offsets(cls, group, topic):
+        return '%s/consumers/%s/offsets/%s' % (cls.KAFKA_CHROOT, group, topic)
+
+    @classmethod
+    def consumer_topic_partition(cls, group, topic, partition):
+        return '%s/%d' % (cls.topic_offsets(group, topic), partition)
+
+    @classmethod
+    def broker_partitions(cls, topic):
+        return '%s/brokers/topics/%s/partitions' % (cls.KAFKA_CHROOT, topic)
+
+    @classmethod
+    def broker_ids(cls):
+        return '%s/brokers/ids' % cls.KAFKA_CHROOT
+
+    @classmethod
+    def broker(cls, broker_id):
+        return '%s/brokers/ids/%s' % (cls.KAFKA_CHROOT, broker_id)
+
+    @classmethod
+    def topics(cls, consumer_group):
+        return '%s/consumers/%s/offsets' % (cls.KAFKA_CHROOT, consumer_group)
+
+
+class KafkaOffsetCollector(object):
```

Comment:
For now, yeah - if we upgrade Kafka beyond 0.8 I would want to look to some other tool like <a href="https://github.com/linkedin/Burrow">Burrow</a> to do offset lag monitoring.